lib/pure/collections/tables.nim

```diff
@@ -1750,9 +1750,56 @@ iterator mvalues*[A, B](t: var OrderedTable[A, B]): var B =
     yield t.data[h].val
     assert(len(t) == L, "the length of the table changed while iterating over it")
 
-
-
-
+type
+  SomeTable*[K, V] = Table[K, V] | OrderedTable[K, V]
```

Comment:
does SomeTable need to be exported?
(if so, it raises questions regarding other tables like CountTable)

Reply:
I exported it to be visible in the documentation that it is union type of `Table` and `OrderedTable`, because it is in the interface of `fromJsonHook` and `toJsonHook` procedures which must be exported. If it is not exported and not visible in the documentation it will be more confusing IMHO.

Reply:
yes, but it's not meant to be used directly, and exporting `SomeTable` leaves CountTable in an awkward place; how about this: get rid of `SomeTable` and inline its definition, eg:
```nim
proc fromJsonHook*[K, V, JN](t: var (Table[K, V] | OrderedTable[K, V]), jsonNode: JN) =
```
i tried, it works

Reply:
Unfortunately, it doesn't work for documentation links.
```nim
##  * `toJsonHook proc<#toJsonHook,(Table[K,V]|OrderedTable[K,V])>`_
```
I tried several variants with parentheses and without, with `Table` only, with spaces around `|` and without.

Maye this is not very important to work correctly or I could remove `See also:` section entirely? With `SomeTable` typedef it still seems cleaner to me. What about if it has to be added also for other table types in the future. The signatures will become too long.

Reply:
you can simply use `toJsonHook` and hopefully docgen will eventually be smarter; it's ok to not have a hyperlink everywhere. The problem with export is it becomes part of the API, and harder to change; it'd be odd to have CountTable singled out.
> What about if it has to be added also for other table types in the future

we can revisit then ;-) (but the other way around is not possible without a breaking change)

Reply:
Fixed.